test/CMakeLists.txt

```diff
@@ -46,10 +47,10 @@ file(GLOB TEST_STUBS stubs/*.c)
 #
 # test_name - the name to call the test
 # test_src - test file that is to be unit tested
-# srcs sources - necessary to build the application, if no external sources
-#                are used, fill this in with the ${DUMMY} variable
-
-function(create_gtest test_name test_src srcs)
+# srcs - list of sources necessary to build the application, if no external sources
+#                are used, fill this in with the ${NONE} variable
+# stubs - list of stubs that are required for the test
```

Comment:
Added a list of stubs to be passed in since the `lib_state_machine_test.cpp` doesn't need bm_os and it was causing a undefined reference error.

Reply:
Interesting, the ll_test does not utilize `bm_os.h` either but does not have this issue. Would you be able to post the build failure?

Reply:
It's likely because ll_test has:
```
#include "fff.h"

DEFINE_FFF_GLOBALS
```
in it.

Reply:
This was the failure:
```
Undefined symbols for architecture arm64:
  "_fff", referenced from:
      _bm_semaphore_create in bm_os.c.o
      _bm_semaphore_create in bm_os.c.o
      _bm_semaphore_give in bm_os.c.o
      _bm_semaphore_give in bm_os.c.o
      _bm_semaphore_take in bm_os.c.o
      _bm_semaphore_take in bm_os.c.o
      _bm_get_tick_count in bm_os.c.o
      _bm_get_tick_count in bm_os.c.o
      ...
```

Reply:
I was able to get it to build ok by adding the following at the top of `lib_state_machine_test.cpp`:
![image](https://github.com/user-attachments/assets/3d371b3b-57f1-42c1-946f-3f5f0bc73213)

Reply:
And I was like wait... how is bm_os being included in this file when it isn't being included in the .h/.c/test.cpp files. 😅

Reply:
I will leave the call up to you although, it would be nice to keep consistency between the `test/CMakeLists.txt` files, as this change would make it functionally different than in `bm_core`

Reply:
Maybe we can add some information in the README for unit tests depending on the route we go to avoid this confusion in the future!

Reply:
I'm leaning towards having a list of sources be passed in as it provides a bit more control + visibilty into what files are being included in each test and removes some unecessary build time for files that are not being used in the test. 

How about something like:
```
#  Linked List (LL) unit tests
set (LL_TEST_SRCS
    # File we are testing
    ${SRC_DIR}/ll.c

    # Supporting files
    ${STUB_DIR}/bm_os.c
)
create_gtest("ll" "${LL_TEST_SRCS}")

#  State Machine unit tests
set (LIB_STATE_MACHINE_TEST_SRCS
    # File we are testing
    ${SRC_DIR}/lib_state_machine.c
)
create_gtest("lib_state_machine" "${LIB_STATE_MACHINE_TEST_SRCS}")
```

I know its pretty verbose but it makes it pretty clear what file is being tested and what is being included to test it.

Regardless I'll add a readme and make sure bm_core tests follow the same flow.

Reply:
Not too verbose, I really like it, nice!